numpy/core/fromnumeric.py

```diff
@@ -1197,7 +1197,17 @@ def resize(a, new_shape):
     See Also
     --------
     ndarray.resize : resize an array in-place.
-
+    
+    Notes
+    -----
+    Warning: This functionality does **not** consider axes separately,
+    i.e. it does not apply interpolation/extrapolation.
+    It fills the return array with the required number of elements,
```

Comment:
Perhaps 'It _instead_ fills the...?

Reply:
In my mind "instead" implies there is some single (better?) alternative, when in fact there are many other alternatives for what `resize` could be doing.

If we were starting over we may have called this `realloc`, since that links back to the memory model this function implements. Over the 20 years since it was created `resize` has taken on another meaning. Perhaps it would be nice to work the word `realloc` into the docstring.

Reply:
Thanks for your comments, let me try to work realloc in the docs.
I meant 'instead' more as in 'this function is not performing interpolation, but INSTEAD doing this and this.' Not trying to make either functionality superior, just highlighting the difference.

Reply:
hmm I might not full understand what you mean with `realloc`, sorry. The returned array of `np.resize()` is a newly created object, not an in-place view right? (i.e changing the resized array does not change the original array.)
Thus `np.resize()` allocates new memory, but doesn't re-allocate anything?

Reply:
correct, but the memory layout is like a `realloc` or a `memcpy`, or `np.ravel(a)` which is actually used internally to implement this. I think what you have is an improvement, maybe we should just use it as-is.

Reply:
The fact `np.ravel()` is being called internally makes a lot of sense yeah, but I'm not sure if including that in the note will make the docs more readable.

If you think it's okay as-is, I hope this pr can get accepted. If not / in the meantime, I'll think about the `realloc`/`np.ravel()` inclusion issue more.